### DISCUSSION TITLE
How to override the commit message when using automerge?

### DISCUSSION THREAD

**Discussion by AlexanderWirthSAP** [renovatebot/renovate#21369] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.31.4

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

3.7.6

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We have enabled automerge for Renovate's pull requests, using the GitHub auto-merge feature.
Our GitHub repository is configured as follows:
- Allow auto-merge: Yes
- Allow merge commits: No
- Allow rebase merging: No
- Allow squash merging: Yes
- Allow squash merging > commit message: Default to pull request title and description

Based on our team's typical workflow, squash merging with the PR title and PR description as the commit message works very well and we want to keep these setting. However, Renovate PRs have a long PR description with lots of verbosity, so when they are used as the commit message they are cluttering up the git history. Therefore I am looking for a way to override the commit message for automerge PRs. **Ideally the commit of Renovate's automerge PRs should only contain the PR title as commit message and nothing else.** Is there any setting that would help me achieve this?

### What you tried so far.

I checked the documentation for relevant settings and I tried setting commitMessagePrefix, commitMessageSuffix, etc. but it changed the PR title instead and the PR description is still copied into the actual commit message of the auto-merged PR.

### Relevant debug logs

N/A

**Comment by rarkins** [comment]
Please provide some current vs desired examples so it's easier to understand

**Reply by AlexanderWirthSAP** [reply]
**Current:**

```
commit 8524bffde3aeb3144d720d950f3bc1317e59ea41 (HEAD -> master, origin/master)
Author: ospo-renovate[bot] <84706+ospo-renovate[bot]@users.noreply.github.wdf.sap.corp>
Date:   Thu Apr 6 14:16:21 2023 +0200

    Update dependency com.sap.sgs:security-patched-spring-boot-starter to v3.0.5-sap-01 (#38)

    This PR contains the following updates:

    | Package | Type | Update | Change |
    |---|---|---|---|
    | [com.sap.sgs:security-patched-spring-boot-starter](https://spring.io/projects/spring-boot) ([source](https://github.wdf.sap.corp/Phosphor/security-patched-spring-boot-30x)) | parent | patch | `3.0.4-sap-01` -> `3.0.5-sap-01` |

    ---

    ### Release Notes

    <details>
    <summary>Phosphor/security-patched-spring-boot-30x</summary>

    ### [`v3.0.5-sap-01`](https://github.wdf.sap.corp/Phosphor/security-patched-spring-boot-30x/blob/HEAD/CHANGELOG.md#&#8203;305-sap-01---2023-03-23)

    ##### Update

    -   Updated to `org.springframework.boot:spring-boot-starter-parent:3.0.5`
        and `org.springframework.boot:spring-boot-dependencies:3.0.5`.

    ##### Security

    -   Updated `org.codehaus.jettison:jettison` from `1.5.2` to `1.5.4`
        to fix [CVE-2023-1436](https://nvd.nist.gov/vuln/detail/CVE-2023-1436).

    ### [`v3.0.4-sap-02`](https://github.wdf.sap.corp/Phosphor/security-patched-spring-boot-30x/blob/HEAD/CHANGELOG.md#&#8203;304-sap-02---2023-03-10)

    ##### Security

    -   Updated `org.wildfly.security:wildfly-elytron` from `1.20.1.Final` to `1.20.3.Final` which inturn updates `org.apache.sshd:sshd-common` from `2.7.0` to `2.9.2`
        to fix [CVE-2022-45047](https://nvd.nist.gov/vuln/detail/CVE-2022-45047).

    </details>

    ---

    ### Configuration

    📅 **Schedule**: Branch creation - At any time (no schedule defined), Automerge - At any time (no schedule defined).

    🚦 **Automerge**: Enabled.

    ♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the rebase/retry checkbox.

    🔕 **Ignore**: Close this PR and you won't be reminded about this update again.

    ---

     - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check this box

    ---

    [Trigger Renovate Run](https://renovate-wdf.internal.opensource.tools.sap/run/LIS/renovate-test) | [View Renovate Logs](https://renovate-wdf.internal.opensource.tools.sap/logs/LIS/renovate-test)
    Generated by [Renovate Bot](https://github.com/renovatebot/renovate) and managed by the [TGIF-project renovate-controller](https://github.tools.sap/OSPO/renovate-controller).
    For questions/issues please raise an issue in the renovate-controller repository.
    <!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNS4zMS40IiwidXBkYXRlZEluVmVyIjoiMzUuMzEuNCJ9-->

    Co-authored-by: OSPO Renovate Bot <84706+ospo-renovate[bot]@users.noreply.github.wdf.sap.corp>
```

**Desired:**

```
commit 8524bffde3aeb3144d720d950f3bc1317e59ea41 (HEAD -> master, origin/master)
Author: ospo-renovate[bot] <84706+ospo-renovate[bot]@users.noreply.github.wdf.sap.corp>
Date:   Thu Apr 6 14:16:21 2023 +0200

    Update dependency com.sap.sgs:security-patched-spring-boot-starter to v3.0.5-sap-01 (#38)
```

**Reply by AlexanderWirthSAP** [reply]
**Github Repo Settings:**

The marked setting is why the whole PR description is included in the commit, but we don't want to change it since it works well for our normal, non-Renovate pull requests. Instead we'd rather tell Renovate via some configuration not to include it in its commit messages.

<img src="https://i.imgur.com/XvcCnV0.png" width="600">

**renovate.json**

```
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "labels": ["renovate"],
    "reviewers": [],
    "timezone": "Europe/Berlin",
    "prConcurrentLimit": 10,
    "prHourlyLimit": 0,
    "separateMinorPatch": true,
    "separateMajorMinor": true,
    "separateMultipleMajor": true,
    "packageRules": [
        {
            "description": "Override default Maven registry.",
            "matchManagers": ["maven"],
            "registryUrls": ["https://int.repositories.cloud.sap/artifactory/build-releases/"]
        },
        {
            "description": "Automatically merge patch changes",
            "matchUpdateTypes": ["patch"],
            "matchLanguages": ["java"],
            "automerge": true,
            "automergeType": "pr",
            "platformAutomerge": true,
            "automergeStrategy": "squash",
            "labels": ["renovate", "automerge"]
        }
    ]
}
```

**Reply by viceice** [reply]
it's not possible for renovate to override that GitHub function

**Reply by rarkins** [reply, marked as answer]
What you're referring to is not Renovate's commit messages, it's GitHub's commit message as a result of automerge. It is in fact Renovate's PR body contents which is going into the automerged commit, not Renovate's commit message.

You can control the prBody to a reasonable extent, but that would mean removing information from each PR which Renovate creates for automerging

**Reply by AlexanderWirthSAP** [reply]
> What you're referring to is not Renovate's commit messages, it's GitHub's commit message as a result of automerge. 

Right, but it's only the default. When you enabled auto-merge on a PR manually, you can change the title and commit message for that merge. And as far as I can tell this is also possible through the API, see the property "commitBody" in [the payload description here](https://docs.github.com/de/graphql/reference/input-objects#enablepullrequestautomergeinput).

So I don't want to remove information from the prBody, I am looking for a way to tell Renovate to set a certain commit body when enabling auto-merge on a PR. Based on your replies, it seems that this is currently not possible. I think it's a reasonable feature request though, given my example of the current behavior above. All I would need is some new setting like for example "automergeCommitBody" that lets me control the commit body that Renovate should use when enabling auto-merge.

**Reply by viceice** [reply]
that sounds good to me. it should be undefined by default to not break existing users.

i would like to have it for Gitea too. 🙃

**Reply by rarkins** [reply]
Sounds good, please create a feature request issue with a summary of the above

**Reply by HonkingGoose** [reply]
I created a feature request for us:

- https://github.com/renovatebot/renovate/issues/21451

**Reply by AlexanderWirthSAP** [reply]
Thank you for creating the feature request, @HonkingGoose
I didn't find the time to do it yet.
